src/api/uploads/BaseUpload.js

```diff
@@ -60,11 +69,51 @@ class BaseUpload extends Base {
         this.xhr.options({
             url,
             data: attributes,
-            successHandler: this.preflightSuccessHandler,
+            successHandler: this.preflightSuccessReachabilityHandler,
             errorHandler: this.preflightErrorHandler,
         });
     };
 
+    /**
+     * Handles successful preflight response.
+     * Performs a upload reachability test before calling preflightSuccessHandler.
+     *
+     * @param {Object} - Request options
+     * @return {Promise} Async function promise
+     */
+    preflightSuccessReachabilityHandler = async ({ data }: { data: { upload_url?: string } }): Promise<any> => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const uploadUrl = data.upload_url;
+        // If uploadUrl is not avaliable, don't make reachability test
+        if (!uploadUrl) {
+            this.preflightSuccessHandler({ data });
+            return;
+        }
+
+        // Parse upload host from uploadUrl
+        const splitUrl = uploadUrl.split('/');
+        const uploadHost = `${splitUrl[0]}//${splitUrl[2]}/`;
+        // The default upload host should always reachable
+        if (uploadHost === `${DEFAULT_HOSTNAME_UPLOAD}/`) {
+            this.preflightSuccessHandler({ data });
+            return;
+        }
+
+        // If upload host reachable upload file, else make a new preflight request
+        const isHostReachable = await this.uploadsReachability.isReachable(uploadHost);
+        if (isHostReachable) {
+            this.preflightSuccessHandler({ data });
+        } else if (this.reachabilityRetryCount >= MAX_REACHABILITY_RETRY) {
+            this.preflightSuccessHandler();
```

Comment:
should we pass data still?

Reply:
See response above ^

Reply:
what does `data` look like? is it just an `uploadUrl` that comes back?

Reply:
Here is a sample of what `data` looks like 
`{
  "upload_token": "Pc3FIOG9vSGV4VHo4QzAyg5T1JvNnJoZ3ExaVNyQWw6WjRsanRKZG5lQk9qUE1BVQP",
  "upload_url": "https://upload-las.app.box.com/api/2.0/files/content?upload_session_id=1234"
}` https://developer.box.com/reference/options-files-content/

Reply:
do we need to use this upload_token to make the reachability test?

Reply:
No, I don't think the upload_token is used any where. I'm not even sure what this token is in the first place.